components/eventing-controller/controllers/subscription/jetstream/reconciler_test.go

```diff
@@ -269,22 +271,44 @@ func TestCreateSubscription(t *testing.T) {
 				K8sSubscription: []gomegatypes.GomegaMatcher{
 					reconcilertesting.HaveCondition(reconcilertesting.DefaultReadyCondition()),
 				},
-				NatsSubscription: []gomegatypes.GomegaMatcher{
-					natstesting.BeExistingSubscription(),
-					natstesting.BeValidSubscription(),
-					natstesting.BeJetStreamSubscriptionWithSubject(reconcilertesting.OrderCreatedEventType),
+				NatsSubscriptions: map[string][]gomegatypes.GomegaMatcher{
+					reconcilertesting.OrderCreatedEventType: {
+						natstesting.BeExistingSubscription(),
+						natstesting.BeValidSubscription(),
+						natstesting.BeJetStreamSubscriptionWithSubject(reconcilertesting.OrderCreatedEventType),
+					},
 				},
 			},
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
+			// The new testing.T instance needs to be used for making assertions. If ignored, errors like this will be printed:
+			// testing.go:1336: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
+			ens := ens
+			ens.G = gomega.NewGomegaWithT(t)
+
+			t.Log("creating the k8s subscription")
 			subscription := utils.CreateSubscription(ens.TestEnsemble, tc.givenSubscriptionOpts...)
+			t.Log("testing the k8s subscription")
 			utils.TestSubscriptionOnK8s(ens.TestEnsemble, subscription, tc.want.K8sSubscription...)
+
+			t.Log("testing the k8s events")
 			utils.TestEventsOnK8s(ens.TestEnsemble, tc.want.K8sEvents...)
-			testSubscriptionOnNATS(ens, subscription, reconcilertesting.OrderCreatedEventType, tc.want.NatsSubscription...)
-			testDeletion(ens, subscription, reconcilertesting.OrderCreatedEventType)
+
+			t.Log("testing the nats subscriptions")
+			for eventType, matchers := range tc.want.NatsSubscriptions {
+				testSubscriptionOnNATS(ens, subscription, eventType, matchers...)
+			}
+
+			t.Log("testing the deletion of the subscription")
+			testSubscriptionDeletion(ens, subscription)
+
+			t.Log("testing the deletion of the NATS subscription(s)")
+			for _, filter := range subscription.Spec.Filter.Filters {
+				ensureNATSSubscriptionIsDeleted(ens, subscription, filter.EventType.Value)
```

Comment:
I see the `ensureNATSSubscriptionIsDeleted` function is always called after the `testSubscriptionDeletion` does it make sense to either merge them, or call the `ensureNATSSubscriptionIsDeleted` inside of `testSubscriptionDeletion` ?